File: backend/app/routers/categories.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.category import Category, CategoryCreate
from app import crud

router = APIRouter(prefix="/api/categories", tags=["categories"])


@router.get("/", response_model=list[Category])
def list_categories(db: Session = Depends(get_db)):
    return crud.category.get_categories


@router.post("/", response_model=Category, status_code=201)
def create_category(category: CategoryCreate, db: Session = Depends(get_db)):
    return crud.category.create_category(db, category)


@router.delete("/{category_id}", response_model=Category)
def delete_category(category_id: int, db: Session = Depends(get_db)):
    db_category = crud.category.delete_category(db, category_id)
    if not db_category:
        raise HTTPException(status_code=404, detail="Category not found")
    return db_category

File: frontend/src/api/categories.js
import { data } from 'autoprefixer'
import api from './axios'

export const categoriesAPI = {
    getAll: () => api.get('/api/categories/').then(r => r.data),
    create: (data) => api.post('/api/categories/', data).then(r => r.data),
    delete: (id) => api.delete(`/api/categories/${id}`).then(r => r.data),
}